### ISSUE TITLE
Remove "ember-cli-release" from the default app blueprint

### ISSUE THREAD

**Issue by Turbo87** [ember-cli/ember-cli#6462]
[`ember-cli-release`](https://github.com/lytics/ember-cli-release) has quite a few large [dependencies](https://github.com/lytics/ember-cli-release/blob/master/package.json#L23-L35) and I personally don't use it at all. Is this a plugin that people commonly use? It seems to me that the community has moved to `ember-cli-deploy` instead, so I propose removing `ember-cli-release` from the default app/addon blueprints. Those of us that use the plugin can just continue to use it, and everyone else will benefit from faster `npm install` times.

/cc @stefanpenner @rwjblue @nathanhammond

**Comment by stefanpenner** [comment]
@slindberg / @lukemelia / @rwjblue  / @elwayman02  you are the top 4 contributors to ember-cli-release, what are your thoughts on:

* can we make this project lighter?
* should this add-on be opt-in rather then default?

**Comment by locks** [comment]
I have to say, as an Ember user I hadn't even realized I was paying the cost of this dependency up-front. I'm in favour of dropping it from the default set of dependencies.

**Comment by nathanhammond** [comment]
Note that I use it in most of my projects, but it should only be there for the addon blueprint. Regardless without looking at it I'm unsure why it pulls in so much, but presumably for good reason.

Waiting on the response from the team responsible for it.

(More generically I'm starting to be of the opinion that if it is included in the default blueprints then it should be a community-owned addon.)

**Comment by kellyselden** [comment]
> (More generically I'm starting to be of the opinion that if it is included in the default blueprints then it should be a community-owned addon.)

👍

**Comment by kellyselden** [comment]
I use this in every addon I do. I have never used it in a regular app. Big +1 for only including it in addons.

**Comment by elwayman02** [comment]
Yea, this seems like it should only be included in addons, and we should probably open an issue on the project itself to investigate if it can be slimmed down (I don't know offhand).

**Comment by stefanpenner** [comment]
> Yea, this seems like it should only be included in addons

Sounds pretty reasonable, @Turbo87 do you share this or..

**Comment by Turbo87** [comment]
@stefanpenner not really... I don't see the value that this addon brings and given that it imports it's own `npm` dependency I don't feel the cost is worth it for the majority of users. From what we've discussed on Slack so far it seems most of what this addon does can be replaced by `npm version` and `git push`.

**Comment by stefanpenner** [comment]
@Turbo87 also sounds reasonable.

So it seems, either it should reduce its dependencies dramatically or we may need to remove it as a default. And if we do include it, it may be an add-on only dependency.

**Comment by kellyselden** [comment]
Hopefully I can give a little perspective. I use this in all of my addons because it was super easy. Then, after I talked to @Turbo87, I found out that `ember release --minor` which I use most often can be replaced with `npm version minor && git push origin master --tags`. So now the addon is not as useful to me as it used to be, although the new command is a little bit more to remember.

I would say though, had I been a less familiar user and this package was missing for a new addon, I would probably be confused, or maybe be irritated about "churn". On the other hand, the docs about releasing an addon make no mention of this addon, it uses raw npm and git commands. So maybe the amount of users this would affect is small.

I think at the very least we should remove from the app blueprint. Then, we should look into condensing the addon, which may mean getting rid of features so the dependencies can go away. If that can't be done cleanly, we should probably drop it.

**Comment by lukemelia** [comment]
+1 on removing from app blueprint. It's rare that I use it with apps and it is as easy to add as any other addon when I do.

I use this for every addon I maintain so I would be +1 keeping it as a dev dependency for addons.

**Comment by Turbo87** [comment]
Could someone please tell me what the addon actually does other then what @kellyselden mentioned above? I still don't see the advantage and if we argue like that we should include ember-cli-deploy in all apps too.

**Comment by lukemelia** [comment]
@Turbo87 The README for ember-cli-release describes its functionality: https://github.com/lytics/ember-cli-release. As you'll see from a read through, It has a number of features and lifecycle hooks beyond version increment, tag and push that community members may be depending on. I've personally used those hooks on past projects to keep version numbers in sync across a couple of related config files.

It sounds like there is agreement in this thread that it should not be included for apps by default.

The estimation of the relative value vs. cost for addons, and the cost to the community of churn, is a judgement call for the core team.

One idea is to implement a helpful error in ember-cli or another addon, that is emitted when a user runs `ember release` after ember-cli-release has been removed, and informs the user `ember-cli-release is not longer installed by default with ember-cli apps/addons. To use it, run 'ember install ember-cli-release'`.

**Comment by elwayman02** [comment]
I personally feel like the default inclusion of this addon has made project
management much easier. Removing it, to me, just adds one more learning
barrier for people who are new to open source. Ember-cli-release is an easy
to understand convenience package; even if you believe it can be replaced
by npm or other tools, having a single command that "just works" is kind of
the point.

**Comment by Turbo87** [comment]
> Removing it, to me, just adds one more learning
> barrier for people who are new to open source.

this feels backwards to me actually. it is yet another thing to learn about when we could limit it to doing it manually or using `npm version` instead.

**Comment by nathanhammond** [comment]
Added to the discussion for 12/1 Ember CLI meeting. Will try to make a call at that point. (Note that this will require an RFC to change.)

**Comment by slindberg** [comment]
> should this add-on be opt-in rather then default?

I have no opinion on this one way or the other. Full disclosure: I'm no longer doing web development, and I have little to no bandwidth to maintain `ember-cli-release` (which is the reason that the latest release is still a `-beta` prerelease). However I do think there's still value in the addon, as explained in the [README](https://github.com/lytics/ember-cli-release#ember-cli-release). I think [@lukemelia's idea to give a helpful warning](https://github.com/ember-cli/ember-cli/issues/6462#issuecomment-263073388) is a good compromise if it is removed.

> can we make this project lighter?

The version that is currently installed with ember-cli is [`v0.2.9`](https://github.com/ember-cli/ember-cli/blob/v2.10.0-beta.2/blueprints/app/files/package.json#L29), whose dependencies if installed by itself weigh in at 6.1Mb. All but 1Mb of that is from `moment` and `moment-timezone`, which are used to support date-based tags. The remaining deps are small and likely to be shared with `ember-cli` itself or other common addons.

Date-based tagging was the original reason I created the addon, but it is probably not used by many people. The addon's tagging strategy was written to be pluggable, so it wouldn't take much to extract it into its own complementary addon (but like I said, I don't have the time to do it).

Note also that the latest version `v1.0.0-beta.2` installs the `npm` package, which is used to publish to NPM if the `--publish` hook is given. This package is a whopping 18Mb (I did not realize this until now), and could fairly easily be removed in favor of shelling out `npm publish`.

**Comment by nathanhammond** [comment]
We discussed this today in the Ember CLI core team meeting in great detail. Here's some color on the conversation that we had.

For new apps and addons simply _not_ including an addon in the default blueprint doesn't break any expectations. The overall API has "changed" but considering it is a brand new project with _incredibly_ low friction to change (`ember install ember-cli-release`) we don't believe it should count as breaking. We don't want to add a "warning" message as a stub command because that is yet another thing we would have to maintain.

For existing apps and addons we already require that people re-`init` the default blueprint in their application. This is their opportunity to elect whether they wish to remove `ember-cli-release`. This results in it being a "backwards compatible" change–at least as backwards compatible as any Ember CLI blueprint change is.

As a general direction we want to get to the point where all addons which we ship by default are owned and maintained under the @ember-cli organization. Even if not explicitly moved to the organization, the test we're applying to new addons that we wish to consider for default inclusion is "are we, Ember CLI Core, willing to maintain this as part of our efforts on Ember CLI?" This is actually a remarkably high bar as we have a limited amount of time and energy and is a significant contributing factor to this decision.

At the end of our conversation we decided that this does _not_ need an RFC as we consider it to be backwards-compatible. We acknowledge nonetheless that it is a change. This may seem antithetical to the intent of our RFC process (discussion of changes, making them community decisions instead of by fiat) but part of the strength of our addon architecture is to make it possible to ship changes like this, safely. We already do a lot of SemVer lawyering where we make a breaking change in a dependency but call it a non-breaking change in Ember CLI, allowing us to move the versions independently. We feel like this falls into that same class of changes. This is not perfect but we hope that you trust _our_ intent to try and do the right thing for the community. We did take into account (and try to rope in the people involved) into this conversation here.

As an additional path forward the @ember-cli team is intending to move all of our addon publishing to be handled by Travis. Manually handling releases is error-prone and something we want to free our users from. As part of that we will be developing Travis scripts to enable this, an effort which you can participate it (more information on this soon). We'd then like to make it possible for all addons to be able to leverage these same scripts to get consistent, repeatable, and automatic releases.

We're very grateful for the work on this addon and very glad to have it as part of the community.

See also @rwjblue's [comment](https://github.com/ember-cli/ember-cli/pull/6507#issuecomment-264305694).

**Comment by elwayman02** [comment]
@nathanhammond can you clarify what the actual resulting decision was, for people who don't want to click through to other PRs/commits to find out what was done?

**Comment by rwjblue** [comment]
TLDR; we removed ember-cli-release from the default app and addon blueprints (in #6507).
